staging/src/k8s.io/metrics/pkg/client/custom_metrics/multi_client.go

```diff
@@ -41,12 +41,13 @@ func PeriodicallyInvalidate(cache AvailableAPIsGetter, interval time.Duration, s
 	ticker := time.NewTicker(interval)
 	defer ticker.Stop()
 
+loop:
 	for {
 		select {
 		case <-ticker.C:
 			cache.Invalidate()
 		case <-stopCh:
-			break
+			break loop
```

Comment:
why?

Reply:
The origin code is just breaking out of the select statement,not achieving the function purpose, that is "periodically invalidates the preferred version cache until told to stop"

Reply:
This change makes sense to me.

Reply:
I think this modify works.

Reply:
I would hope so, it's your PR! 🤣